nntrainer/tensor/manager.h

```diff
@@ -299,9 +302,11 @@ class Manager {
       deallocateWeights();
 
     if (tensors_allocated) {
-      deallocateGradients();
+      if (model_trainable)
+        deallocateGradients();
       deallocateInOuts();
-      deallocateDerivatives();
+      if (model_trainable)
```

Comment:
how about `model_training`? if the model is trainable this boolean can still false.

Reply:
done.